java-frontend/src/main/java/org/sonar/java/ast/JavaAstScanner.java

```diff
@@ -64,14 +66,18 @@ public void scan(Iterable<? extends InputFile> inputFiles) {
     boolean successfullyCompleted = false;
     boolean cancelled = false;
     try {
+      ExecutionTimeReport executionTimeReport = new ExecutionTimeReport(Clock.systemUTC());
       for (InputFile inputFile : inputFiles) {
         if (analysisCancelled()) {
           cancelled = true;
           break;
         }
+        executionTimeReport.start(inputFile.uri().getPath());
         simpleScan(inputFile);
+        executionTimeReport.end();
```

Comment:
I feel that this call to `end()` and probably a call to `report()` should be part of the finally block as well, in case of interruption.

Reply:
Good point, in case of a never-ending analysis, it could be nice to see the report. I'm moving the `report()` in the finally block and make it works when `end()` is missing.